test/jdk/java/nio/channels/AsynchronousFileChannel/Basic.java

```diff
@@ -561,22 +562,76 @@ static void testTruncate(Path file) throws IOException {
         }
     }
 
+    // tests unsupported MemorySegment view buffers
+    static void testViewsOfUnsupportedArenas(Path file) throws IOException {
+        System.out.println("testViewsOfUnsupportedArenas");
+
+        AsynchronousFileChannel ch = AsynchronousFileChannel
+            .open(file, CREATE, READ, WRITE, TRUNCATE_EXISTING);
+
+        writeFully(ch, genBuffer(), 0L);
+        long size = ch.size();
+
+        try {
+            readAll(ch, genUnsupportedBuffer(true), 0L);
+            throw new RuntimeException("IllegalStateException expected");
+        } catch (IllegalStateException expected) {
+            // ignore
+        }
+
+        try {
+            readAll(ch, genUnsupportedBuffer(false), 0L);
+            throw new RuntimeException("UnsupportedOperationException expected");
+        } catch (UnsupportedOperationException expected) {
+            // ignore
+        }
+
+        try {
+            writeFully(ch, genUnsupportedBuffer(true), size);
+            throw new RuntimeException("IllegalStateException expected");
+        } catch (IllegalStateException expected) {
+            // ignore
+        }
+
+        try {
+            writeFully(ch, genUnsupportedBuffer(false), size);
+            throw new RuntimeException("UnsupportedOperationException expected");
+        } catch (UnsupportedOperationException expected) {
+            // ignore
+        } finally {
+            ch.close();
+        }
+    }
+
     // returns ByteBuffer with random bytes
     static ByteBuffer genBuffer() {
         int size = 1024 + rand.nextInt(16000);
         byte[] buf = new byte[size];
-        return switch (rand.nextInt(3)) {
+        rand.nextBytes(buf);
+        return switch (rand.nextInt(4)) {
             case 0 -> ByteBuffer.allocateDirect(buf.length)
                     .put(buf)
                     .flip();
             case 1 -> ByteBuffer.wrap(buf);
-            case 2 -> Arena.ofAuto().allocate(buf.length).asByteBuffer()
+            case 2 -> Arena.global().allocate(buf.length).asByteBuffer()
+                    .put(buf)
+                    .flip();
+            case 3 -> Arena.ofAuto().allocate(buf.length).asByteBuffer()
                     .put(buf)
                     .flip();
             default -> throw new InternalError("Should not reach here");
         };
     }
 
+    // returns ByteBuffer view of confined or shared arena
+    static ByteBuffer genUnsupportedBuffer(boolean isConfined) {
+        int size = 1024 + rand.nextInt(16000);
+        byte[] buf = new byte[size];
+        rand.nextBytes(buf);
+        Arena arena = isConfined ? Arena.ofConfined() : Arena.ofShared();
+        return arena.allocate(buf.length).asByteBuffer().put(buf).flip();
```

Comment:
For the asynchronous channels then we should only prohibit thread-confined, using a buffer that is a view of a memory segment allocated from a shared arena is okay.  I've included (disabled-for-now) test in draft changes for JDK-8358958 to test attempting to close a shared arena while a write op is outstanding, I think we can use that to test the changes here.